torch_xla/distributed/xla_backend.py

```diff
@@ -0,0 +1,289 @@
+        f'World size: {world_size}, slice length: {slice_len}')
+
+  # ensure the ranks are a correct range.
+  start = slice_ranks[0]
+  step = slice_ranks[1] - slice_ranks[0]
+  stop = start + step * slice_len
+  expected_ranks = list(range(start, stop, step))
+  if slice_ranks != expected_ranks:
+    raise ValueError(
+        f'Given slice isn\'t a range with evenly distributed steps: {slice_ranks}'
+    )
+
+  num_slices = int(world_size / slice_len)
+  world = list(range(world_size))
+  mesh = []
+  if step == 1:
+    # Horizontal case.
+    if start % slice_len != 0 or start >= world_size:
+      raise ValueError('Given horizontal slice doesn\'t have a correct start: '
+                       f'World size: {world_size}, slice length: {slice_len}, '
+                       f'slice: {slice_ranks}')
+    for i in range(num_slices):
+      slice_start = i * slice_len
+      slice_stop = (i + 1) * slice_len
+      slice_idx = slice(slice_start, slice_stop, step)
+      mesh.append(world[slice_idx])
+  else:
+    # Vertical case.
+    if start >= num_slices:
+      raise ValueError('Given vertical slice doesn\'t have a correct start: '
+                       f'World size: {world_size}, slice length: {slice_len}, '
+                       f'slice: {slice_ranks}')
+    if step != num_slices:
+      raise ValueError('Given vertical slice doesn\'t have a correct step: '
+                       f'World size: {world_size}, slice length: {slice_len}, '
+                       f'slice: {slice_ranks}')
+    for i in range(num_slices):
+      slice_start = i
+      slice_stop = i + slice_len * step
+      slice_idx = slice(slice_start, slice_stop, step)
+      mesh.append(world[slice_idx])
+  assert slice_ranks in mesh
+
+  return mesh
+
+
+def new_xla_process_group(ranks=None,
+                          timeout=dist.default_pg_timeout,
+                          backend=None):
+  pg = _orig_new_group_fn(ranks, timeout, backend)
+  if isinstance(pg, ProcessGroupXla) and ranks is not None:
+    world_pg = dist.group.WORLD
+    if not isinstance(world_pg, ProcessGroupXla):
+      raise RuntimeError('xla backend requires the default ProcessGroup to be '
+                         'a ProcessGroupXla')
+
+    if isinstance(ranks, range):
+      ranks = list(ranks)
+
+    if ranks == list(range(world_pg.size())):
+      pg._mesh = [ranks]
+    elif len(ranks) == 1:
+      if ranks[0] not in range(world_pg.size()):
+        raise ValueError('Given ranks is out of range: '
+                         f'World size: {world_pg.size()}, ranks: {ranks}')
+      pg._mesh = [[r] for r in range(world_pg.size())]
+    elif len(ranks) < world_pg.size() and len(ranks) > 1:
+      pg._mesh = infer_mesh(ranks, world_pg.size())
+    else:
+      logging.warn(
+          f'Can\'t infer process group mesh from given ranks "{str(ranks)}". '
+          'The process group will use the entire world as its collective comm group.'
+      )
+
+  return pg
```

Comment:
looks like if `ranks == None`, this function will return the original `pg`. This would work correctly. However, we have seen users creating multiple PG instances to bump up network utilization. Do we expect such use cases here as well?

Reply:
This was not considered. What would be the potential problem if using the original `pg` in such a case?

Reply:
sorry, I thought I deleted this comment before submitting reviews. Please ignore.